packages/core/contracts/financial-templates/funding-rate-store/implementation/FundingRateStore.sol

```diff
@@ -238,43 +283,139 @@ contract FundingRateStore is FundingRateStoreInterface, Testable, Lockable {
         address proposer = disputeSucceeded
             ? fundingRateDispute.proposal.disputer
             : fundingRateDispute.proposal.proposer;
+        emit DisputedRateSettled(
+            perpetual,
+            proposalTime,
+            fundingRateDispute.proposal.proposer,
+            fundingRateDispute.proposal.disputer,
+            disputeSucceeded
+        );
 
         IERC20 collateralCurrency = IERC20(PerpetualInterface(perpetual).getCollateralCurrency());
+        // TOOD: Decide whether loser of dispute should lose entire bond or partial
         if (disputeSucceeded) {
             // If dispute succeeds:
-            // - pay disputer the dispute reward + final fee rebate
+            // - Disputer earns back their bonds: dispute bond + final fee bond
+            // - Disputer earns as reward: the proposal bond
+            FixedPoint.Unsigned memory disputerRebate = fundingRateDispute.proposal.proposalBond.add(
```

Comment:
Question: Should we store an additional `disputeBond` in the `fundingRateRecord`? Currently I do NOT because the `disputeBond == proposalBond`.

Reply:
pros of keeping it like this is it's simpler. cons are that we are bound to keeping these bonds equal. there could be a scenario where we want them to be different but I think for now it's fine to assume that they are equal.

Reply:
Yeah I think if we want to make them different in the future we can just add the param to the struct